plugins/source/azure/README.md

```diff
@@ -0,0 +1,127 @@
+
+```bash
+az login
+```
+
+Then, create the service principal cloudquery will use to access your cloud deployment. WARNING: The output of
+`az ad sp create-for-rbac` contains credentials that you must protect - Make sure to handle with appropriate care.
+This example uses bash - The commands for CMD and PowerShell are similar.
+
+```bash
+export SUBSCRIPTION_ID=<YOUR_SUBSCRIPTION_ID>
+az account set --subscription $SUBSCRIPTION_ID
+az provider register --namespace 'Microsoft.Security'
+
+# Create a service-principal for cloudquery
+az ad sp create-for-rbac --name cloudquery-sp --scopes /subscriptions/$SUBSCRIPTION_ID --role Reader
+```
+
+(you can, of course, choose any name you'd like for your service-principal, `cloudquery-sp` is just an example.
+If the service principal doesn't exist it will create a new one, otherwise it will update an existing one)
+
+The output of `az ad sp create-for-rbac` should look like this:
+
+```
+{
+  "appId": <YOUR AZURE_CLIENT_ID>,
+  "displayName": "cloudquery-sp",
+  "password": <YOUR AZURE_CLIENT_SECRET>,
+  "tenant": <YOUR AZURE_TENANT_ID>
+}
+```
+
+**Exporting environment variables**
+
+Next, you need to export the environment variables that cloudquery will use to `fetch` your cloud configuration.
+Copy them from the output of `az ad sp create-for-rbac` (or, take the opportunity to show off your jq-foo).
+The example shows how to export environment variables for Linux - exporting for CMD and PowerShell is similar.
+
+- `AZURE_TENANT_ID` is `tenant` in the JSON.
+- `AZURE_CLIENT_ID` is `appId` in the JSON.
+- `AZURE_CLIENT_SECRET` is `password` in the JSON.
+
+```bash
+export AZURE_TENANT_ID=<YOUR AZURE_TENANT_ID>
+export AZURE_CLIENT_ID=<YOUR AZURE_CLIENT_ID>
+export AZURE_CLIENT_SECRET=<YOUR AZURE_CLIENT_SECRET>
+export AZURE_SUBSCRIPTION_ID=$SUBSCRIPTION_ID
+```
+
+### Configuration
+
+The following configuration section can be automatically generated by `cloudquery gen source azure`:
+
+```yaml title="azure.yml"
+kind: source
+spec:
+  name: azure
+  tables: ["*"]
+  destinations: ["postgresql"]
+  spec:
+    # configuration:
+    # subscriptions:
+    # Optional: If you only want to fetch data from a single or multiple subscriptions, you can specify it here.
+    # - "<subscription_1>"
+    # - "<subscription_2>"
+```
+
+By default, CloudQuery will fetch all configurations from **all** resources in **all** subscriptions. You can change this behaviour with the following arguments:
+
+- `subscriptions` **(Optional)** - Specify multiple subscriptions to fetch data from them concurrently.
+
+### Query Examples
+
+#### Find all MySQL servers
+
+```sql
+SELECT * FROM azure_mysql_servers;
+```
+
+#### Find storage accounts that are allowing non-HTTPS traffic
+
+```sql
+SELECT * from azure_storage_accounts where enable_https_traffic_only = false;
+```
+
+#### Find all expired key vaults
+
+```sql
+SELECT * from azure_keyvault_vault_keys where attributes_expires >= extract(epoch from now()) * 1000;
```

Comment:
```suggestion
SELECT * from azure_keyvault_keys where attributes_expires >= extract(epoch from now()) * 1000;
```

And we don't unwrap `attributes` so we need a JSON path query.
So maybe change this to simpler example or remove it